File: PayJunction C#/ConsoleApplication2/PayJunction/Transaction/TransactionRequest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using RestSharp;
using Newtonsoft.Json;

namespace PayJunction.Transaction
{
    class NewTransactionRequest : Request<TransactionResponse>
    {
        public enum Mode
        {
            Charge,
            Refund
        }

        /// <summary>
        /// Defaults to Charge
        /// </summary>
        public Mode RequestMode
        {
            set
            {
                Params["action"] = value == Mode.Charge ? "CHARGE" : "REFUND";
            }
        }


        /// <summary>
        /// Endpoint. Don't include slash in beginning.
        /// </summary>
        public override string EndPoint
        {
            get
            {
                return "transactions";
            }
        }

        public string CreditCardNumber
        {
            set
            {
                Params["cardNumber"] = value;
            }get
            {
                return Params["cardNumber"];
            }
        }

        public string CardExpiryMonth
        {
            set
            {
                Params["cardExpMonth"] = value;
            }
            get
            {
                return Params["cardExpMonth"];
            }
        }

        public string CardExpiryYear
        {
            set
            {
                Params["cardExpYear"] = value;
            }
            get
            {
                return Params["cardExpYear"];
            }
        }

        /// <summary>
        /// ID of terminal to run transaction on.
        /// </summary>
        public string TerminalID
        {
            get
            {
                return Params["terminalId"];
            }
            set
            {
                Params["terminalId"] = value;
            }
        }

        /// <summary>
        /// Empty to turn off cvv check. Else CVV number
        /// </summary>
        public string CardCVV
        {
            set
            {
                Params["cvv"] = String.IsNullOrEmpty(value) ? "OFF" : "ON";
                Params["cardCvv"] = value;
            }
            get
            {
                return Params["cardCvv"];
            }
        }

        public string InvoiceNumber
        {
            get
            {
                return Params["invoiceNumber"];
            }
            set
            {
                Params["invoiceNumber"] = value;
            }
        }

        public string AmountBase
        {
            get
            {
                return Params["amountBase"];
            }
            set
            {
                Params["amountBase"] = value;
            }
        }

        public override Method Method
        {
            get
            {
                return Method.POST;
            }
        }

        public NewTransactionRequest(string cardNumber, string expYear, string expMonth,string amount)
        {
            this.CardExpiryMonth = expMonth;
            this.CardExpiryYear = expYear;
            this.CreditCardNumber = cardNumber;
            this.AmountBase = amount;
            this.CardCVV = "";
            this.RequestMode = Mode.Charge;
        }

        /// <summary>
        /// It creates new response everytime. Caller you reuse it instead of calling it multiple time
        /// </summary>
        /// <returns></returns>
        public override TransactionResponse ProcessResponse()
        {
            TransactionResponse res = JsonConvert.DeserializeObject<TransactionResponse>(RawResponse);
            return res;
        }
    }
}

File: PayJunction Webhook/PayJunction/PayJunction/Transaction/TransactionRequest.vb
Imports Newtonsoft
Imports Newtonsoft.Json
Imports Newtonsoft.Json.Linq
Imports RestSharp

Namespace PayJunction.Transaction
    Class NewTransactionRequest
        Inherits Request(Of TransactionResponse)
        Public Enum Mode
            Charge
            Refund
        End Enum

        ''' <summary>
        ''' Defaults to Charge
        ''' </summary>
        Public WriteOnly Property RequestMode() As Mode
            Set
                Params("action") = If(Value = Mode.Charge, "CHARGE", "REFUND")
            End Set
        End Property


        ''' <summary>
        ''' Endpoint. Don't include slash in beginning.
        ''' </summary>
        Public Overrides ReadOnly Property EndPoint() As String
            Get
                Return "transactions"
            End Get
        End Property

        Public Property CreditCardNumber() As String
            Get
                Return Params("cardNumber")
            End Get
            Set
                Params("cardNumber") = Value
            End Set
        End Property

        Public Property CardExpiryMonth() As String
            Get
                Return Params("cardExpMonth")
            End Get
            Set
                Params("cardExpMonth") = Value
            End Set
        End Property

        Public Property CardExpiryYear() As String
            Get
                Return Params("cardExpYear")
            End Get
            Set
                Params("cardExpYear") = Value
            End Set
        End Property

        ''' <summary>
        ''' ID of terminal to run transaction on.
        ''' </summary>
        Public Property TerminalID() As String
            Get
                Return Params("terminalId")
            End Get
            Set
                Params("terminalId") = Value
            End Set
        End Property

        ''' <summary>
        ''' Empty to turn off cvv check. Else CVV number
        ''' </summary>
        Public Property CardCVV() As String
            Get
                Return Params("cardCvv")
            End Get
            Set
                Params("cvv") = If([String].IsNullOrEmpty(Value), "OFF", "ON")
                Params("cardCvv") = Value
            End Set
        End Property

        Public Property InvoiceNumber() As String
            Get
                Return Params("invoiceNumber")
            End Get
            Set
                Params("invoiceNumber") = Value
            End Set
        End Property

        Public Property AmountBase() As String
            Get
                Return Params("amountBase")
            End Get
            Set
                Params("amountBase") = Value
            End Set
        End Property

        Public Overrides ReadOnly Property Method() As Method
            Get
                Return Method.POST
            End Get
        End Property

        Public Sub New(cardNumber As String, expYear As String, expMonth As String, amount As String)
            Me.CardExpiryMonth = expMonth
            Me.CardExpiryYear = expYear
            Me.CreditCardNumber = cardNumber
            Me.AmountBase = amount
            Me.CardCVV = ""
            Me.RequestMode = Mode.Charge
        End Sub

        ''' <summary>
        ''' It creates new response everytime. Caller you reuse it instead of calling it multiple time
        ''' </summary>
        ''' <returns></returns>
        Public Overrides Function ProcessResponse() As TransactionResponse
            Dim res = JsonConvert.DeserializeObject(Of TransactionResponse)(RawResponse)
            Return res
        End Function
    End Class
End Namespace